Pidaq/Lib/uploadToDrive/uploadToDrive.py

```diff
@@ -0,0 +1,125 @@
+        parentFolderID = DefaultID
+        logging.error('Folder name not found \nuploading to "Default"')
+    
+    #Today's date in YYYY-MM-DD format is stored in the variable "date"
+    date = datetime.today().strftime('%Y-%m-%d')
+    
+    # If the folder does not exist, should be the case
+    # when running for the first time on a given day
+    if checkExistingSubFolder(date, parentFolderID) == 0:
+        folder_metadata = {
+            'name' : date,
+            'mimeType': 'application/vnd.google-apps.folder',
+            'parents' : [parentFolderID]
+            }
+        # Creates a folder with the given date
+        folder = DRIVE.files().create(body = folder_metadata,
+                                      fields = 'id').execute()
+        # Returns the date folder's ID
+        folderId = getFolderId(date, parentFolderID)
+    # executes if the folder already exists  
+    elif checkExistingSubFolder(date, parentFolderID) == 1:
+        logging.warning("found folder for today's date")
+        folderId = getFolderId(date, parentFolderID)
+    # executes if more than one folder exists
+    else:
+        logging.warning('unexpected count of folders named with a given date')
+        return
+    
+    # extracts the .csv file name from the path
+    fileName=os.path.basename(filePath)
+    metadata = {'name': fileName,
+                'parents' : [folderId]
+                }
+    # uploads the .csv file
+    response = DRIVE.files().create(body=metadata,
+                               media_body=filePath).execute()
+    
+    if response:
+        logging.warning('Uploaded "%s" (%s)' % (metadata['name'], response['mimeType']))
+    else:
+        logging.warning('error in uploading the csv file')
+        
+def checkExistingSubFolder(folderName, parentFolder):
+    """ Queries the drive for the folder with
+        the given name and parent folder
+    """
+    query = " name = '" + folderName + "' and parents = \"" + parentFolder +'"'
+    response = DRIVE.files().list(q = query).execute()
+
+    # Returns 0 if no folder exists with today's date
+    # in the specified parent folder
+    if len(response.get('files',[])) == 0: 
+        logging.warning("creating a new folder...")
+        return 0
+    # Returns 1 if the folder with today's date
+    # exists in the specified parent folder
+    elif len(response.get('files',[])) == 1: 
+        for data in response.get('files', []):
+            return 1
+    else:
+        logging.warning("More than one file found")
+
+def getFolderId(fName, parentFolder):
+    """ Performs the query to return the ID of the folder
+        with the given name
+    """
+    # searches in the drive using the parameters passed
+    query = " name = '" + fName + "' and parents = \"" + parentFolder +'"'
+    response = DRIVE.files().list(q = query).execute()
+    # Gets the folder ID if it exists
+    if len(response.get('files',[])) == 1:
+        for data in response.get('files', []):

```

Comment:
Is this loop necessary? you check that the length is equal to 1, so there is nothing to loop through. This could just be something like:
```python
return response.get('files', []).get('id')
```